components/support/nimbus-fml/src/defaults/validator.rs

```diff
@@ -161,23 +213,23 @@ impl<'a> DefaultsValidator<'a> {
         Ok(())
     }
 
-    fn validate_types(&self, path: &ErrorPath, type_ref: &TypeRef, default: &Value) -> Result<()> {
+    fn validate_types(
+        &self,
+        path: &ErrorPath,
+        type_ref: &TypeRef,
+        default: &Value,
+        errors: &mut Vec<FeatureValidationError>,
+    ) {
         match (type_ref, default) {
             (TypeRef::Boolean, Value::Bool(_))
             | (TypeRef::BundleImage, Value::String(_))
             | (TypeRef::BundleText, Value::String(_))
             | (TypeRef::String, Value::String(_))
             | (TypeRef::StringAlias(_), Value::String(_))
             | (TypeRef::Int, Value::Number(_))
-            | (TypeRef::Option(_), Value::Null) => Ok(()),
+            | (TypeRef::Option(_), Value::Null) => (),
             (TypeRef::Option(inner), v) => {
-                if let TypeRef::Option(_) = inner.as_ref() {
-                    return Err(FMLError::ValidationError(
-                        path.path.to_string(),
-                        "Nested options".into(),
-                    ));
-                }
```

Comment:
This is redundant (it's being [detected in the `schema/validator.rs` already](https://github.com/mozilla/application-services/commit/11b6f771d8763bf338b552e1d8fb2487ef95444a#diff-543c056cb67d81785681f221906751a7412e66a2009fe65202dceec84aff2302R180-R185)).